docs/DeveloperGuide.adoc

```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
+
+==== Design Considerations
+
+===== Aspect: Implementation of data structure
+
+* **Alternative 1 (current choice):** Doing the calculation only when the user wants to view the statistics.
+** Pros: Calculating only when needed is more efficient.
+** Cons: The command would take some time to calculate before being able to display the statistics.
+
+* **Alternative 2:** Every change to the database updates the statistics on the fly.
+** Pros: Displaying the statistics on the UI would be faster.
+** Cons: Need to change the implementation of every command that causes a change to the database. Calculations have to be done every operation, affecting efficiency.
+
+* Decision: Alternative 1 was adopted as it is an easier implementation. As the number of times the `stats` command is called is likely to be low as compared to an `add` or `delete` operation, doing the calculation after every operation would be a waste of computation power. 
+
+===== Aspect: Design of UI display
+
+* **Alternative 1 (current choice):** Displaying the statistics in a new window.
+** Pros: The statistics would be displayed independently from the database and would not interfere with each other. 
+** Cons: The user can open multiple windows which contains different data and not know which is the correct window.
+
+* **Alternative 2:** Hide the displayed database to show the statistics.
+** Pros: There would be only 1 window to manage at any point in time.
+** Cons: Cannot view the statistics and the database concurrently.
+
+* Decision: Alternative 1 was adopted as it is an easier implementation. Implementing the `stats` window as a static window that will not change would not affect the other parts of the app, and there is no need to hide and show the current screen. Moreover, the user might want to view the statistics together with the database, which is only possible with multiple windows.
+
+=== The panel UI for `Member`, `Staff`, `Booking`, and `Ingredient`
+
+The restaurant book displays the entire database in the form of "cards", which is the same as AB4. Each item has its own panel.
+
+=== Current implementation
+
+The panel is formatted in the `ItemListPanel.fxml` file, and each item has its own controller (`MemberListPanel`, `StaffListPanel`, `BookingListPanel`, and `IngredientListPanel`) that extends from `ItemListPanel`. The panel displays the database via cards. Each item has its own card (`MemberCard`, `StaffCard`, `BookingCard`, and `IngredientCard`).
+
+==== Design Considerations
+
+===== Aspect: Design of card display
+
+* **Alternative 1 (current choice):** Making a separate fxml file for each item card.
```

Comment:
For each item card vs for each type of item card

If I have 100 members in the restaurant-> 100 fxml files?